exporter/stackdriverexporter/resource_mapper_test.go

```diff
@@ -38,45 +43,78 @@ func TestResourceMapper(t *testing.T) {
 		wantResource   *monitoredres.MonitoredResource
 	}{
 		{
-			name: "Converted resource with matching labels",
+			name: "Converted resource with all matching labels",
 			sourceResource: &resource.Resource{
 				Type: "source.resource1",
 				Labels: map[string]string{
-					"source.label1": "value1",
+					"renamedLabel": "value1",
 					"contrib.opencensus.io/exporter/stackdriver/project_id": "123",
+					"unknown": "value1",
 				},
 			},
 			wantResource: &monitoredres.MonitoredResource{
 				// Resource type transformed
 				Type: "target_resource_1",
 				Labels: map[string]string{
-					// One of the labels transformed
-					"target_label_1": "value1",
-					"contrib.opencensus.io/exporter/stackdriver/project_id": "123",
+					// Both labels transformed
+					"project_id":   "123",
+					"target_label": "value1",
 				},
 			},
 		},
 		{
-			name: "Converted resource without matching labels",
+			name: "Converted resource with optional label missing",
 			sourceResource: &resource.Resource{
-				Type: "source.resource2",
+				Type: "source.resource1",
+				Labels: map[string]string{
+					"renamedLabel": "value1",
+					"unknown":      "value1",
+				},
+			},
+			wantResource: &monitoredres.MonitoredResource{
+				// Resource type transformed
+				Type: "target_resource_1",
+				Labels: map[string]string{
+					// Required label transformed
+					"target_label": "value1",
+				},
+			},
+		},
+		{
+			name: "Converted resource with required label missing",
+			sourceResource: &resource.Resource{
+				Type: "source.resource1",
 				Labels: map[string]string{
-					"source.label1": "value1",
 					"contrib.opencensus.io/exporter/stackdriver/project_id": "123",
 				},
 			},
+			// Resource with missing required labels should be converted via default implementation
 			wantResource: &monitoredres.MonitoredResource{
 				// Resource type transformed
-				Type: "target_resource_2",
+				Type: "global",
 				Labels: map[string]string{
-					// Labels passed through with no changes
-					"source.label1": "value1",
+					"project_id": "123",
```

Comment:
For a moment I was wondering why project_id got converted even though the other mandatory label is missing. But the project_id is converted by default implementation.  Perhaps add this info in the comment.

Reply:
Added comments to clarify.